lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -365,12 +365,21 @@ private void onContainerStatus(final ContainerStatus value) {
       case COMPLETE:
         LOG.log(Level.FINE, "Container completed: status {0}", value.getExitStatus());
         switch (value.getExitStatus()) {
-        case 0:
+        case ContainerExitStatus.SUCCESS:
           status.setState(ReefServiceProtos.State.DONE);
           break;
-        case 143:
+        case ContainerExitStatus.KILLED_BY_RESOURCEMANAGER:
+        case ContainerExitStatus.KILLED_EXCEEDED_PMEM:
+        case ContainerExitStatus.KILLED_EXCEEDED_VMEM:
           status.setState(ReefServiceProtos.State.KILLED);
           break;
+        case ContainerExitStatus.PREEMPTED:
+          status.setState(ReefServiceProtos.State.PREEMPTED);
+          break;
+        case ContainerExitStatus.ABORTED:
+        case ContainerExitStatus.DISKS_FAILED:
+        case ContainerExitStatus.KILLED_AFTER_APP_COMPLETION:
+        case ContainerExitStatus.KILLED_BY_APPMASTER:
         default:
```

Comment:
I've actually left that out on purpose as that shouldn't happen according to the YARN protocol. But you're right, leaving just that one out is kind of awkward. I've re-added it and also left a comment in the code.